src/main/java/tds/exam/results/messaging/ExamCompletedMessageListener.java

```diff
@@ -42,8 +45,23 @@ public ExamCompletedMessageListener(final ExamResultsService examResultsService,
      */
     public void handleMessage(final String examId) {
         LOG.debug("Received completed exam notification for id: {}", examId);
+        if (properties.isRetryOnError()) {
+            processMessage(examId);
+        } else {
+            try {
+                processMessage(examId);
+            } catch (final Exception e) {
+                // Log error, and do not rethrow to prevent ERT from re-processing this same request
+                LOG.error("Error occurred while processing or sending the exam results for examId {}: {}", examId,
+                    e);
+            }
+        }
```

Comment:
Moving the retry checking up from the ExamResultsService into here also catches problems with the call to examService.updateStatus as well as handles a malformed message payload that may not be resolvable via UUID.fromString.